src/arch/x86/reg/id.rs

```diff
@@ -1,5 +1,50 @@
+use crate::arch::RegId;
+
 // TODO: Add proper `RegId` implementation. See [issue #29](https://github.com/daniel5151/gdbstub/issues/29)
 // pub enum X86RegId {}
 
-// TODO: Add proper `RegId` implementation. See [issue #29](https://github.com/daniel5151/gdbstub/issues/29)
-// pub enum X86_64RegId {}
+/// 64-bit x86 core + SSE register identifier.
+///
+/// Source: https://github.com/bminor/binutils-gdb/blob/master/gdb/features/i386/64bit-core.xml
+/// Additionally: https://github.com/bminor/binutils-gdb/blob/master/gdb/features/i386/64bit-sse.xml
+#[derive(Debug, Clone, Copy)]
+#[non_exhaustive]
+pub enum X86_64CoreRegId {
+    /// General purpose registers: RAX, RBX, RCX, RDX, RSI, RDI, RBP, RSP,
+    /// r8-r15
+    Gpr(u8),
+    /// Instruction pointer
+    Rip,
+    /// Status register
+    Eflags,
+    /// Segment registers: CS, SS, DS, ES, FS, GS
+    Segment(u8),
+    /// FPU registers: ST0 through ST7
+    St(u8),
+    /// FPU internal registers: FCTRL, FSTAT, FTAG, FISEG, FIOFF, FOSEG, FOOFF,
+    /// FOP
+    Fpu(u8),
```

Comment:
Instead of a `u8`, could we instead have a separate `X87FpuInternalRegId` enum which corresponds to the `X87FpuInternalRegs`?

I don't think it'd need it's own `impl RegId`, as the implementation might differ between `x86` and `x86_64`.

Reply:
Done. Instead of implementing `RegId`, I used `FromPrimitive` to convert an integer to an enum value.